packages/recommend-react/src/__tests__/useFrequentlyBoughtTogether.test.ts

```diff
@@ -24,22 +26,62 @@ function createMockedRecommendClient() {
 }
 
 describe('useFrequentlyBoughtTogether', () => {
-  test('gets Frequently Bought Together products', () => {
+  test('returns FBT recommendations', async () => {
     const { recommendClient } = createMockedRecommendClient();
 
-    renderHook(() => {
-      const { recommendations } = useFrequentlyBoughtTogether({
+    const { result } = renderHook(() =>
+      useFrequentlyBoughtTogether({
         indexName: 'test',
         recommendClient,
         threshold: 0,
         objectIDs: ['testing'],
         queryParameters: {
           facetFilters: ['test'],
         },
-        transformItems: (items) => items,
-      });
+      })
+    );
+    await waitFor(() => {
+      expect(result.current.recommendations).toEqual([hit]);
+    });
+  });
+
+  test('assures that the transformItems function always returns an array', async () => {
+    const { recommendClient } = createMockedRecommendClient();
+
+    const { result, rerender } = renderHook(
+      () =>
+        useFrequentlyBoughtTogether({
+          indexName: 'test',
+          recommendClient,
+          threshold: 0,
+          objectIDs: ['testing'],
+          queryParameters: {
+            facetFilters: ['test'],
+          },
+          transformItems: (items) => {
+            return items.map((item) => {
+              return item.name;
+            });
+          },
+        }),
+      {
+        wrapper: StrictMode,
+      }
+    );
+    await waitFor(() => {
+      expect(result.current.recommendations).toEqual([
+        'Landoh 4-Pocket Jumpsuit',
+      ]);
+    });
+
+    act(() => {
+      rerender();
```

Comment:
This tests that the weird array behavior is gone, but it doesn't test that the feature actually works.

Here, we want to rerender with another `transformItems` reference, and to make sure that the items have been transformed with this latest function reference.